src/pubkey.h

```diff
@@ -205,4 +210,13 @@ struct CExtPubKey {
     bool Derive(CExtPubKey& out, unsigned int nChild) const;
 };
 
+/** Users of this module must hold an ECCVerifyHandle. The constructor and
+ *  destructor of these are not allowed to run in parallel, though. */
```

Comment:
That would be highly inefficient. Not every user needs his own context. The
purpose of ECCVerifyHandle is to ensure that the single global context
variable is always initialized while someone is using it. You can't have a
refcount variable per user, you need a single refcount for all users.